### PR TITLE
Log error instead of crashing the game in job_coroutines.lua

### LINKED ISSUE #20: Error in job_thread: attempt to get length of field 'nodedata' (a nil value)
MineTest 5.4.1

ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod 'working_villages' in callback luaentity_Step(): ...ds\working_villages\working_villagers/job_coroutines.lua:42: error in job_thread ...mods\working_villages\working_villagers/jobs/builde
2021-08-02 11:47:11: ERROR[Main]: r.lua:41: attempt to get length of field 'nodedata' (a nil value)
2021-08-02 11:47:11: ERROR[Main]: stack traceback:
2021-08-02 11:47:11: ERROR[Main]: 	[C]: in function 'error'
2021-08-02 11:47:11: ERROR[Main]: 	...ds\working_villages\working_villagers/job_coroutines.lua:42: in function 'resume'
2021-08-02 11:47:11: ERROR[Main]: 	...4\bin\..\mods\working_villages\working_villagers/api.lua:929: in function <...4\bin\..\mods\working_villages\working_villagers/api.lua:914>

### PR DESCRIPTION
**Pull request by ketwaroo**
fixes the root cause of many crashes that point to same line.

TL;DR

`error()` throws an exception basically and causes everything to crash. Logging the error, especially if it's not critical, seems the more sensible thing to do.

Long explanation:

at least 3 different errors all boil down to same line job_corutine.lua:42

fixes #20 amongst other things. 

A lot of other mods and even the default game assume that actions are performed by the player only and do not expect villagers to be interacting with certain nodes and entities. It is virtually impossible to debug each and every one of them or add extra code in this mod to account for it. For example #39 adds further logic to remember failed nodes but if every node fails, we run the risk of out of memory issues as the list of failed nodes grows and grows.

While it seems tempting to throw an error to force the user to report the issue, it just makes the average user think that this mod is broken and they will more likely just uninstall it.

### CODE REVIEW

**Comment by ketwaroo** [comment]
Closing since 1721deabe7fbaca6b952b067d8943297e56e8ca3 provides a better solution
